docs/src/main/tut/docs/cassandra/README.md

```diff
@@ -4,6 +4,265 @@ title: Cassandra
+
+```
+@free
+trait ClusterAPI {
+
+  def connect: FS[Session]
+
+  def connectKeyspace(keyspace: String): FS[Session]
+
+  def close: FS[Unit]
+
+  def configuration: FS[Configuration]
+
+  def metadata: FS[Metadata]
+
+  def metrics: FS[Metrics]
+
+}
+``` 
+
+### SessionAPI
+Provides a way to interact with a proper query. We can both define a query as a template - containing 
+placeholders for real values - or a raw query string. Bear in mind that running a raw query using the 
+SessionAPI directly is unsafe and does not check query correctness at compile time. 
+
+```
+@free
+trait SessionAPI {
+
+  def init: FS[Session]
+
+  def close: FS[Unit]
+
+  def prepare(query: String): FS[PreparedStatement]
+
+  def prepareStatement(statement: RegularStatement): FS[PreparedStatement]
+
+  def execute(query: String): FS[ResultSet]
+
+  def executeWithValues(query: String, values: Any*): FS[ResultSet]
+
+  def executeWithMap(query: String, values: Map[String, AnyRef]): FS[ResultSet]
+
+  def executeStatement(statement: Statement): FS[ResultSet]
+
+  def executeWithByteBuffer(
+      query: String,
+      values: List[SerializableValueBy[Int]],
+      consistencyLevel: Option[ConsistencyLevel] = None): FS[ResultSet]
+
+}
+```
+
+### StatementAPI
+Provides methods to bind real query values to an already existing PreparedStatement, returning a 
+BoundStatement, which can now be ran in a safe way.
+
+```
+@free
+trait StatementAPI {
+
+  def bind(preparedStatement: PreparedStatement): FS[BoundStatement]
+
+  def setByteBufferByIndex(
+      boundStatement: BoundStatement,
+      index: Int,
+      bytes: ByteBuffer): FS[BoundStatement]
+
+  def setByteBufferByName(
+      boundStatement: BoundStatement,
+      name: String,
+      bytes: ByteBuffer): FS[BoundStatement]
+
+  def setValueByIndex[T](
+      boundStatement: BoundStatement,
+      index: Int,
+      value: T,
+      codec: ByteBufferCodec[T]): FS[BoundStatement]
+
+  def setValueByName[T](
+      boundStatement: BoundStatement,
+      name: String,
+      value: T,
+      codec: ByteBufferCodec[T]): FS[BoundStatement]
+
+  def setByteBufferListByIndex(
+      preparedStatement: PreparedStatement,
+      values: List[SerializableValueBy[Int]]): FS[BoundStatement]
+
+  def setByteBufferListByName(
+      preparedStatement: PreparedStatement,
+      values: List[SerializableValueBy[String]]): FS[BoundStatement]
+}
+```
+
+### ResultSetAPI
+Provides methods to interact directly with a Cassandra ResultSet, so we can get the final Object 
+representation, a list of them, or describe that that ResultSet could not have been found in the 
+current Cassandra instance.
+
+```
+@free
+trait ResultSetAPI {
+
+  def read[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[A]
+
+  def readOption[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[Option[A]]
+
+  def readList[A](resultSet: ResultSet)(implicit FR: FromReader[A]): FS[List[A]]
+
+}
+```
+
+## Freatures
+
+### Low level Queries
```

Comment:
Low level API? 

Then we might have different sections for either DDL or DML

Reply:
Thanks @juanpedromoreno! At the moment we are only supporting DDL ops. I'm gonna clarify this to avoid confussions. Thanks a lot!

Reply:
Thanks @juanpedromoreno! At the moment we are supporting both DDL and DML ops, being them transparent to the final user. We are describing here how to interact with datastax object at a low level. I'm not sure if I'm getting you here, sorry. Thanks a lot!